helix-core/src/main/java/org/apache/helix/manager/zk/zookeeper/ZkClient.java

```diff
@@ -1754,4 +1784,64 @@ private void recordStateChange(boolean stateChanged, boolean dataChanged) {
       }
     }
   }
+
+  /**
+   * Creates a {@link org.apache.helix.manager.zk.zookeeper.IZkStateListener} that wraps a default
+   * implementation of {@link org.I0Itec.zkclient.IZkStateListener}, which means the returned
+   * listener runs the methods of {@link org.I0Itec.zkclient.IZkStateListener}.
+   * This is for backward compatibility with {@link org.I0Itec.zkclient.IZkStateListener}.
+   */
+  private static class IZkStateListenerI0ItecImpl implements IZkStateListener {
```

Comment:
nit, The logic is totally fine, but the fact that the simular code is split into different places concerns me a little bit. Do you think it would be cleaner if we merge the 2 default implementation and put it close to the interface def? It's a question, I'm not sure if that will really make it easier to understand.

Reply:
Thanks for the suggestion. I actually had the same concern when implementing this class. I tried to put this class in the interface, but realized that ZkClient is not in the same package with the interface. If we want to access this implantation class, we have to make it public in the interface, which is not what I want.

If I don't understand your point correctly, please correct me.

Reply:
Makes sense. Let's leave it here for now.